yt_dlp/extractor/forendors.py

```diff
@@ -0,0 +1,268 @@
+import functools
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    InAdvancePagedList,
+    clean_html,
+    format_field,
+    int_or_none,
+    unified_strdate,
+    unified_timestamp,
+)
+from ..utils.traversal import traverse_obj
+
+
+class ForendorsBaseIE(InfoExtractor):
+    _NETRC_MACHINE = 'forendors'
+    _API_BASE = 'https://api.forendors.cz'
+    _BASE_URL = 'https://www.forendors.cz'
+
+    @functools.cached_property
+    def _api_headers(self):
+        """Get API headers with CSRF token from cookies"""
+        csrf_token = self._get_cookies(self._API_BASE).get('XSRF-TOKEN')
+        headers = {
+            'Accept': 'application/json, text/plain, */*',
+            'Referer': f'{self._BASE_URL}/',
+        }
+        if csrf_token:
+            headers['x-xsrf-token'] = urllib.parse.unquote(csrf_token.value)
+        return headers
+
+    def _ensure_csrf_token(self, url, page_id):
+        """Ensure CSRF token cookie is set by requesting the page if needed"""
+        if not self._get_cookies(self._API_BASE).get('XSRF-TOKEN'):
+            self._request_webpage(url, page_id, note='Requesting CSRF token cookie')
+
+    def _extract_thumbnails(self, post):
```

Comment:
huh? Your tests never extract a thumbnail, you want to pass the video component and not the post.
also should be simplified using traverse_obj and use url_or_none to check that the URLs are valid.